src/main/blackbox/blackbox.c

```diff
@@ -1706,19 +1709,29 @@ void blackboxUpdate(timeUs_t currentTimeUs)
 }
 
 /*
- * Returns start time in ISO 8601 format, YYYY-MM-DDThh:mm:ss
- * Year value of "0000" indicates time not set
+ * Formats start time in ISO 8601 format, YYYY-MM-DDThh:mm:ss
+ * Year value of "0000" indicates time not set.
  */
-static char startDateTime[20] = "0000-01-01T00:00:00";
-const char *blackboxGetStartDateTime(void)
+
+static char started_buf[FORMATTED_DATE_TIME_BUFSIZE];
+const char * blackboxGetStartDateTime()
 {
-    return startDateTime;
+    return started_buf;
 }
 
-void blackboxSetStartDateTime(const char *dateTime, timeMs_t timeNowMs)
+void blackboxSaveStartDateTime()
 {
-    (void)dateTime;
-    (void)timeNowMs;
+    date_time_t dt;
+    if (!rtc_get_dt(&dt)) {
+        dt.year = 0;
+        dt.month = 1;
+        dt.day = 1;
+        dt.hours = 0;
+        dt.minutes = 0;
+        dt.seconds = 0;
+        dt.nanos = 0;
+    }
+    date_time_format(started_buf, &dt);
```

Comment:
If `rtc_get_dt` will return false `date_time_format` will format a random date/time - `dt` struct is allocated on stack and not zeroed.

Reply:
If `rtc_get_dt()` returns false, the code inside the `if` sets the year to zero, month to 1, etc... This doesn't format a random value, this makes the code return 0000-01-01... when there's no time available, just like it did before. I think this should be kept as is.

Reply:
Oops, I obviously overlooked that.

Reply:
On the other hand it may be desired to move that reature to RTC so `rtc_get_dt` will always provide valid data - return `false` will merely indicate that there's no time data available.